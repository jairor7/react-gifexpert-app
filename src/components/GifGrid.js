// import React, { useState, useEffect } from "react";
import React from "react";
import PropTypes from "prop-types";
import { useFetchGifs } from "../hooks/useFechGifs";
import GifGridItem from "./GifGridItem";

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loading && <p>Loading...</p>}
      <div className="container container-cards">
        {images.map((img) => (
          <GifGridItem {...img} key={img.id} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
