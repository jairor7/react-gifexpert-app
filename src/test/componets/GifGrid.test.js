import GifGrid from "../../components/GifGrid";
import { shallow } from "enzyme";
import { useFetchGifs } from "../../hooks/useFechGifs";
jest.mock("../../hooks/useFechGifs");

describe("Pruebas del componente <GifGrid />", () => {
  const category = "One Punch";

  test("Debe de mostrar el componente GifGrid", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar items cuando se recibe data del useFetchGifs", () => {
    //Mock
    const gifs = [
      { id: 1, title: "Title 1", url: "localhost/1.jpg" },
      { id: 2, title: "Title 2", url: "localhost/2.jpg" },
      { id: 3, title: "Title 3", url: "localhost/3.jpg" },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
