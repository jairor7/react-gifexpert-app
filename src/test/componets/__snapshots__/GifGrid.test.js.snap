// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifGrid /> Debe de mostrar el componente GifGrid 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="container container-cards"
  />
</Fragment>
`;

exports[`Pruebas del componente <GifGrid /> Debe de mostrar items cuando se recibe data del useFetchGifs 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <div
    className="container container-cards"
  >
    <GifGridItem
      id={1}
      key="1"
      title="Title 1"
      url="localhost/1.jpg"
    />
    <GifGridItem
      id={2}
      key="2"
      title="Title 2"
      url="localhost/2.jpg"
    />
    <GifGridItem
      id={3}
      key="3"
      title="Title 3"
      url="localhost/3.jpg"
    />
  </div>
</Fragment>
`;
